geniza/corpus/views.py

```diff
@@ -75,20 +77,27 @@ def get_queryset(self):
 
         self.queryset = documents
 
-        # return 50 documents for now; pagination TODO
-        return documents[:50]
+        return documents
 
-    def get_context_data(self):
-        context_data = super().get_context_data()
-        # should eventually be handled by paginator, but
-        # patch in total number of results for display for now
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+
+        paginator = Paginator(self.queryset, per_page=self.paginate_by)
+        page_num = self.request.GET.get("page", 1)
+        current_page = paginator.page(page_num)
```

Comment:
per django documentation, paginator object and current page object should already be set in context data; let's use those

https://docs.djangoproject.com/en/3.2/ref/class-based-views/mixins-multiple-object/#django.views.generic.list.MultipleObjectMixin.get_context_data